packages/server/database/types/processTeamsLimitsJob.ts

```diff
@@ -0,0 +1,38 @@
+import ms from 'ms'
+import {r} from 'rethinkdb-ts'
+import sendTeamsLimitEmail from '../../billing/helpers/sendTeamsLimitEmail'
+import {DataLoaderWorker} from '../../graphql/graphql'
+import isValid from '../../graphql/isValid'
+import ScheduledTeamLimitsJob from './ScheduledTeamLimitsJob'
+
+const processTeamsLimitsJob = async (job: ScheduledTeamLimitsJob, dataLoader: DataLoaderWorker) => {
+  const {orgId, runAt, type} = job
+  const [organization, orgUsers] = await Promise.all([
+    dataLoader.get('organizations').load(orgId),
+    dataLoader.get('organizationUsersByOrgId').load(orgId)
+  ])
+  const {name: orgName, scheduledLockAt, lockedAt} = organization
+
+  if (!scheduledLockAt || lockedAt) return
+
+  const expectedRunAt =
+    type === 'LOCK_ORGANIZATION' ? scheduledLockAt.getTime() : scheduledLockAt.getTime() - ms('7d')
+
+  if (expectedRunAt !== runAt.getTime()) return
```

Comment:
@igorlesnenko I think I wrote this following your example - I think we were checking if the `runAt` time is the same as `expectedRunAt` because the scheduled job may have been updated, e.g. if the user upgrades, but wouldn't we cancel the scheduled job instead of changing the `runAt` timestamp? I'll remove this bit for now

Reply:
@nickoferrall Currently we don't cancel Jobs if user upgrades or fixes the limit. Shall we? 

This check might also prevent race condition:

- Job fetched from DB
- User upgraded
- Job executed

Thought it is a very rare case

What do you think?

Reply:
@igorlesnenko yeah I think we'll need to cancel the job if the user upgrades, no? What currently happens to the job if the user upgrades?

Reply:
@nickoferrall Currently, nothing happens to the job if user upgrades. I thought that `expectedRunAt` check would "cancel" job when it starts. But we can remove job in a real time when user upgrades or deletes a team.

Reply:
As in this check: `if (expectedRunAt !== runAt.getTime()) return`?

Does the `expectedRunAt` date change when they upgrade? It seems more intuitive to me that the job is removed once the user upgrades, but I'm open to other ideas.

I've created an issue to capture this: https://github.com/ParabolInc/parabol/issues/7698